sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Utils.java

```diff
@@ -51,16 +58,16 @@ static String getEtag(String cosmosDbPayload) {
     }
 
     private static <T> Document<T> parseDocument(JsonObject obj, Class<T> documentType) {
-        T document = sGson.fromJson(obj.get(Constants.DOCUMENT_FIELD_NAME), documentType);
         try {
+            T document = sGson.fromJson(obj.get(Constants.DOCUMENT_FIELD_NAME), documentType);
             return new Document<>(
                     document,
                     obj.get(Constants.PARTITION_KEY_FIELD_NAME).getAsString(),
                     obj.get(Constants.ID_FIELD_NAME).getAsString(),
                     obj.has(Constants.ETAG_FIELD_NAME) ? obj.get(Constants.ETAG_FIELD_NAME).getAsString() : "",
                     obj.get(Constants.TIMESTAMP_FIELD_NAME).getAsLong());
         } catch (Exception exception) {
```

Comment:
why is the exception a checked one here? Isn't that RuntimeException?

Reply:
Change made